File: dkplayer-java/src/main/java/xyz/doikki/videoplayer/render/TextureViewRenderFactory.java
package xyz.doikki.videoplayer.render;

import android.content.Context;

public class TextureViewRenderFactory implements RenderFactory {

    public static TextureViewRenderFactory create() {
        return new TextureViewRenderFactory();
    }

    @Override
    public Render create(Context context) {
        return new TextureViewRender(context);
    }
}

File: dkplayer-java/src/main/java/xyz/doikki/videoplayer/render/TextureViewRenderFactory.kt
package xyz.doikki.videoplayer.render

import android.content.Context

internal class TextureViewRenderFactory : RenderFactory {

    override fun create(context: Context): Render {
        return TextureViewRender(context)
    }

    override fun equals(other: Any?): Boolean {
        println("equals ${other is TextureViewRenderFactory}")
        if (other is TextureViewRenderFactory)
            return true
        return super.equals(other)
    }

    override fun hashCode(): Int {
        println("hashCode $this ${this.javaClass.hashCode()}")
        return this.javaClass.hashCode()
    }
}